app/models/miq_db_config.rb

```diff
@@ -1,221 +0,0 @@
-class MiqDbConfig
-  include Vmdb::Logging
-  @@common_options = [
-    # note sure if this is required (blank == localhost)
-    {:name => :host,     :description => "Hostname",          :required => false},
-    {:name => :database, :description => "Database Name",     :required => true},
-    {:name => :username, :description => "Username",          :required => true},
-    {:name => :password, :description => "Password",          :required => false},
-  ]
-  @@common_fields = @@common_options.collect { |o| o[:name] }
-
-  @@db_types = [
-    {:name => "internal",     :description => "Internal Database on this CFME Appliance"},
-    {:name => "external_evm", :description => "External Database on another CFME Appliance"},
-    {:name => "postgresql",   :description => "External Postgres Database"}
-  ]
-
-  @@defaults = {
-    :adapter      => "postgresql",
-    :username     => "root",
-    :database     => "vmdb_production",
-    :encoding     => "utf8",
-    :pool         => 5,
-    :wait_timeout => 5
-  }
-
-  @@cfg_dir  = File.join(Rails.root, "config")
-  @@db_yml   = File.join(@@cfg_dir, "database.yml")
-
-  @@pwd_mask = {:password => "[PASSWORD]", :verify => "[PASSWORD]"}
-
-  DB_FILE = File.join(Rails.root, "data/db_settings")
```

Comment:
This is one less user of the [data directory](https://github.com/ManageIQ/manageiq/blob/904caa27384f6451b3a128f6528f88d3d6269fe6/app/models/miq_server.rb#L117-L121)...I think the only one left is in https://github.com/ManageIQ/manageiq/blob/904caa27384f6451b3a128f6528f88d3d6269fe6/lib/vmdb/util.rb#L101, and maybe that can just use the Rails tmp directory?